framework/base/Module.php

```diff
@@ -571,32 +571,43 @@ public function createController($route)
             return false;
         }
 
-        if (strpos($route, '/') !== false) {
-            list($id, $route) = explode('/', $route, 2);
-        } else {
-            $id = $route;
-            $route = '';
+        // module and controller map take precedence
+        $routeInfo=pathinfo($route);
+        $id =$routeInfo['dirname']!=='.'?$routeInfo['dirname']:$routeInfo['basename'];
+        $route = $routeInfo['dirname']!=='.'?$routeInfo['basename']:'';
+        if (isset($this->controllerMap[$id.'/'.$route])) {
+            $controllerClass= $this->controllerMap[$id.'/'.$route];
+            $id=$id.'/'.$route;
```

Comment:
Doesn't `$route` already contain ID?

Reply:
No. Though the parameter `$route` has always been  given as `moduleId/controllerId/actionId`, the returned  variable `$route` in `[$controller, $route]` has been always  the `actionId`. I am sorry it was also challenging for me with these variable naming.

In fact this is where the issue #13433 is solved. 

Thanks